docs/kubernetes/overlays.md

```diff
@@ -0,0 +1,125 @@
+   ```
+
+   For `stable-1-19`, 
+   ```
+    apiVersion: kustomize.config.k8s.io/v1beta1
+    kind: Kustomization
+    namespace:
+    gce-pd-csi-driver
+    resources:
+    - ../../base/controller
+    - ../../base/node_linux
+    patchesStrategicMerge:
+    - <your-1-19-kustomize-diff-patch.yaml>
+    transformers:
+    - ../../images/stable-1-19
+   ```
+
+   For `prow-gke-release-staging-rc-1-19`,
+   ```
+    apiVersion: kustomize.config.k8s.io/v1beta1
+    kind: Kustomization
+    resources:
+    - ../stable-1-19
+    transformers:
+    - ../../images/prow-gke-release-staging-rc-1-19
+   ```
+
+   * If the changes are long term, make the changes to `deploy/kubernetes/base`, add the appropriate diff to remove the changes from the overlays less than `stable-1-19`. The overlay kustomize would look something like this
+
+   For `stable-1-19`,
+   ```
+    apiVersion: kustomize.config.k8s.io/v1beta1
+    kind: Kustomization
+    namespace:
+    gce-pd-csi-driver
+    resources:
+    - ../../base
+    # no patches needed since changes merged to deploy/kubernetes/base
+    transformers:
+    - ../../images/stable-1-19
+   ```
+
+   For `prow-gke-release-staging-rc-1-19`,
+   ```
+    apiVersion: kustomize.config.k8s.io/v1beta1
+    kind: Kustomization
+    resources:
+    - ../stable-1-19
+    transformers:
+    - ../../images/prow-gke-release-staging-rc-1-19
+   ```
+
+   For `stable-1-18` and older overlays,
+   ```
+    apiVersion: kustomize.config.k8s.io/v1beta1
+    kind: Kustomization
+    namespace:
+    gce-pd-csi-driver
+    resources:
+    - ../../base
+    patchesStrategicMerge:
+    - <your-1-18-kustomize-diff-remove-1-19-changes-patch.yaml>
+    transformers:
+    - ../../images/stable-1-18
+   ```
+
+3. Driver image update
+   * If we have released a new driver image version, update the `deploy/kubernetes/images/prow-gke-release-staging-rc-{k8s-minor}/image.yaml` and `deploy/kubernetes/images/prow-gke-release-staging-rc-master/image.yaml`, for whichever k8s minor version are compatible with the new driver image version.
+   * Validate the staging drivers and then make changes to the `deploy/kubernetes/images/stable-{k8s-minor}/image.yaml`and `deploy/kubernetes/images/stable-master/image.yaml`
+
+4. Cutting new releases
+   * With the master branch maintaining the stable-x.y overlays, the master branch stable overlays can always be the referenced to pick the latest and greatest stable driver manifest bundle.
+   * At the time of cutting a release branch (e.g release-1.0), we should ensure all necessary manifests are cherry-picked appropriately to the newly cut branch. This ensures that the release branch has a valid overlays manifest bundle, and tests running against the release branch will work. As the master branch moves forward, the release branch is still valid (but not necessarily up-to-date). For the most up-to-date manifest bundle, refer to the master stable-x.y overlays.
```

Comment:
What are typically the "necessary manifests" to be cherrypicked? In the design doc you mentioned "Update the rc overlays for the k8s minor versions which the driver supports."

Reply:
This wording is confusing. I wanted to convey to keep the overlays for the supported k8s minor version valid. Changed the wording now